Update getRecentPrioritizationFees json-rpc docs (#29666)
solana-labs/solana: docs/src/developing/clients/jsonrpc-api.md

### getRecentPrioritizationFees

> **UNSTABLE**

Reviewer: Practically I don't how flagging this method as unstable matters. Once clients adopt it, and we really hope they will quickly, it's going to be very tough to make non-backwards compatible changes to it.

Author: Hmm, yeah, that's fair. I am hoping any future changes are additive regardless.

Returns a list of minimum prioritization fees from recent blocks. Currently, a
node's prioritization-fee cache stores data from up to 150 blocks.

#### Parameters:

> - `<array>` - (optional) An array of account address strings (up to a maximum of 128 addresses). If this parameter is provided, the response will reflect the minimum prioritization fee to land a transaction locking all of the provided accounts as writable.

Reviewer 2: looks like we accept loading up to 256 accounts per transaction via ALT (which is probably reckless, I thought it was 64)

https://github.com/solana-labs/solana/blob/69c4db27c58ce133ff0b611bac723b992ec9d325/sdk/program/src/message/versions/v0/mod.rs#L133

Reviewer 2: nm, it's still capped at load to 64 in `Bank::get_transaction_account_lock_limit`

Author: Are you suggesting we change this? https://github.com/solana-labs/solana/blob/56f8c164975a2b68762e7fe22057955bcacdefd3/rpc/src/rpc.rs#L4004

Because `MAX_TX_ACCOUNT_LOCKS` is definitely set to 128

Reviewer 2: `MAX_TX_ACCOUNT_LOCKS` is only in use on devnet.  64 is hardcoded on the other clusters 'til the feature is activated

https://github.com/solana-labs/solana/blob/56f8c164975a2b68762e7fe22057955bcacdefd3/runtime/src/bank.rs#L3628

pretty great that we have three live values implying that they do the same thing

Author: > `MAX_TX_ACCOUNT_LOCKS` is only in use on devnet.

Why is that feature active on devnet and not testnet? 💔 
Well, anyway, `MAX_TX_ACCOUNT_LOCKS` is what the `getRecentPrioritizationFees` RPC is using, so this doc is correct. Let's create an issue elsewhere, if that value should be changed.

#### Results:

An array of:

- `RpcPrioritizationFee<object>`
  - `slot: <u64>` - Slot in which minimum fee was observed

>   - `prioritizationFee: <u64>` - Minimum fee paid for a successfully landed transaction, specified in increments of 0.000001 lamports

Reviewer: ```suggestion
  - `prioritizationFee: <u64>` - a fee paid by at least one successfully landed transaction, specified in increments of 0.000001 lamports
```

Author: Should I strip out the other references to minimum? I do so in a 2nd commit so you can see how it looks.

### getRecentPrioritizationFees

Returns a list of prioritization fees from recent blocks. Currently, a
node's prioritization-fee cache stores data from up to 150 blocks.

#### Parameters:

> - `<array>` - (optional) An array of account address strings (up to a maximum of 128 addresses). If this parameter is provided, the response will reflect a fee to land a transaction locking all of the provided accounts as writable.

Reviewer: The naked array input here is actually a little unfortunate too. One day read-only accounts (or even executables) may affect prioritization as well. I guess the input becomes an array *or* an object

Author: Yeah, that's true. Sigh, I should know better by now, and just use an input config object for any and all new endpoints.
Fortunately, I have lots of practice with the enum shenanigans to fix.
